Makefile

```diff
@@ -44,10 +44,24 @@ LDFLAGS := -X github.com//open-telemetry/opentelemetry-ebpf-profiler/vc.version=
 	-X github.com/open-telemetry/opentelemetry-ebpf-profiler/vc.buildTimestamp=$(BUILD_TIMESTAMP) \
 	-extldflags=-static
 
-GO_FLAGS := -buildvcs=false -ldflags="$(LDFLAGS)" -tags osusergo,netgo
+# Valid values: release, debug
+BUILD_TYPE ?= release
```

Comment:
We can get rid of this completely and inline the debug build changes into the debug target, this avoids a recursive `MAKE` call

```suggestion
```